inc/Engine/CriticalPath/DataManager.php

```diff
@@ -0,0 +1,135 @@
+<?php
+namespace WP_Rocket\Engine\CriticalPath;
+
+use WP_Error;
+
+/**
+ * Class DataManager
+ *
+ * @package WP_Rocket\Engine\CriticalPath
+ */
+class DataManager {
+
+	/**
+	 * Base critical CSS path for posts.
+	 *
+	 * @var string
+	 */
+	private $critical_css_path;
+
+	/**
+	 * DataManager constructor, adjust the critical css path for posts.
+	 *
+	 * @param string $critical_css_path path for main critical css folder.
+	 */
+	public function __construct( $critical_css_path ) {
+		$this->critical_css_path = $critical_css_path . get_current_blog_id() . DIRECTORY_SEPARATOR;
```

Comment:
Add a `$filesystem` property and set it here in the constructor.

```php
$this->filesystem = rocket_direct_filesystem();
```

Why? It's being used over and over again. Each time we invoke `rocket_direct_filesystem()`, it takes memory and time to instantiate a new `WP_Direct_Filesystem` object. We can invoke once, store in a property in the object, and then use the property throughout the object.

Note: When we build a `Filesystem` module, then we can inject it as a dependency. We could inject _now_ in the ServiceProvider.

Reply:
I think for now we can't do that because of the following:-
1. Plugin tries to call load method here https://github.com/wp-media/wp-rocket/blob/4f40095d616ad3e3f485d2580e037e731b5bb3a6/inc/main.php#L39-L40
2. Which loads the service provider for cpcss https://github.com/wp-media/wp-rocket/blob/253905f6ad3207b32a182ec67410e75e8b2f3b7a/inc/classes/class-plugin.php#L70
3. And inside the service provider it tries to get an instance of DataManager class which inside constructor call the function `rocket_direct_filesystem`
https://github.com/wp-media/wp-rocket/blob/253905f6ad3207b32a182ec67410e75e8b2f3b7a/inc/Engine/CriticalPath/ServiceProvider.php#L51
4. Then after that it tries to load files.php which contains the definition for the function `rocket_direct_filesystem` https://github.com/wp-media/wp-rocket/blob/4f40095d616ad3e3f485d2580e037e731b5bb3a6/inc/main.php#L44

So it will fail with `Fatal error: Uncaught Error: Call to undefined function`

Reply:
We can. We need to move where `files.php` is being loaded. We want to load that right after composer's autoloader is loaded. Then it's available everyone. Let me create a PR for that to get it into `develop`. Then we can merge it here.

Reply:
Great, now we can :+1: